src/Sulu/Bundle/AdminBundle/DependencyInjection/Compiler/AddPropertyMetadataMapperPass.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\DependencyInjection\Compiler;
+
+use Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata\SelectionPropertyMetadataMapper;
+use Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata\SingleSelectionPropertyMetadataMapper;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+class AddPropertyMetadataMapperPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $fieldTypeOptionRegistry = $container->get('sulu_admin.field_type_option_registry');
+        $baseFieldTypes = $fieldTypeOptionRegistry->toArray();
```

Comment:
Or can we maybe add a priority to the compiler pass? Would that even be a better solution than doing this in the `SuluAdminExtension`?

Reply:
It would solve the problem of not all options being added (unless somebody registers a compiler pass with an even lower priority).

But I think it is still not a good idea... It opens the door to errors like we recently had, which caused the `cache:clear` command to fail because there was no database configured. Calling `$container->get` is already calling the constructor, which might not be ready to be called. This might be fine for now, but who know what happens with the `FieldTypeOptionRegistry` in the future, maybe this will break at some point...

Reply:
Okay, I'm also fine with doing this in the extension